src/sentry/services/hybrid_cloud/lostpasswordhash.py

```diff
@@ -0,0 +1,67 @@
+import datetime
+from abc import abstractmethod
+from dataclasses import dataclass
+
+from sentry.models import BaseLostPasswordHash, LostPasswordHash
+from sentry.services.hybrid_cloud import (
+    CreateStubFromBase,
+    InterfaceWithLifecycle,
+    silo_mode_delegation,
+)
+from sentry.silo import SiloMode
+
+
+@dataclass(frozen=True)
+class APILostPasswordHash(BaseLostPasswordHash):
```

Comment:
Does this `BaseLostPasswordHash` work in practice?  I ran into numerous mypy issues doing the `BaseUser`.

Also.  I just don't like the name.  `Base` is way, too... meaningless.  It also conflicts with other things we happen to name `Base`, inculding `BaseUser`, etc..

Reply:
I renamed it LostPasswordHashMixin. This one isn't type checked anywhere, so we shouldn't have mypy problems. I literally just want it for the shared method(s), e.g. `get_absolute_url`